### PR TITLE
Avoids returning an error if sharing a direct channel fails after its creation

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR updates the create direct channel logic to only log an error if the channel was not shared successfully. New direct channels are the only ones inspected to be auto-shared right now, so only them are affected by this bug.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-58019

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/channel.go [hunk 1]
```diff
@@ -481,7 +481,7 @@ func (a *App) createDirectChannelWithUser(c request.CTX, user, otherUser *model.
 		}
 
 		if _, err := a.ShareChannel(c, sc); err != nil {
-			return nil, model.NewAppError("CreateDirectChannel", "app.sharedchannel.dm_channel_creation.internal_error", nil, "", http.StatusInternalServerError).Wrap(err)
+			c.Logger().Error("Failed to share newly created direct channel", mlog.String("channelId", channel.Id), mlog.Err(err))
```
Comment (lieut-data) [inline server/channels/app/channel.go:484]: Nit: not sure of server conventions, but can we use snake case for logging parameters? Makes it easier when consuming Loki.

**Review by lieut-data — APPROVED** [review]

File: server/channels/app/channel.go (hunk 1 above)
Comment (mgdelacroix) [reply server/channels/app/channel.go]: Sure, fixed in https://github.com/mattermost/mattermost/pull/26895/commits/08aa33a6e5642d90fbbe5d693b70bb8efb0b327d

**Review by sbishel — APPROVED** [review]

**Pull request merged**